nion/swift/ImageCanvasItem.py

```diff
@@ -1073,15 +1110,66 @@ def _update_image_canvas_position(self, widget_delta: Geometry.FloatSize) -> Geo
             self._set_image_canvas_position(new_image_canvas_position)
         return new_image_canvas_position
 
+    def convert_pixel_to_normalised(self, coord: Geometry.IntPoint) -> Geometry.FloatPoint:
+        if coord:
+            widget_mapping = ImageCanvasItemMapping.make(self.__data_shape, self.__composite_canvas_item.canvas_bounds,
+                                                         list())
+            if widget_mapping:
+                mapped = self.map_widget_to_image(coord)
+                if (mapped is not None) and (self.__data_shape is not None):
+                    norm_coord = tuple(ele1 / ele2 for ele1, ele2 in zip(iter(mapped), iter(self.__data_shape)))
```

Comment:
Can you name these "y"  instead of "ele1" and "x"  instead of "ele2" ? I'm not sure what "ele" means but it is confusing to a casual code reviewer.

Reply:
y and x in this context wouldn't be anymore descriptive (and potentially confusing) as they are not the X and Y parts.  I have renamed them to something hopefully a bit more descriptive (mapped_coord and shape_dim) to try represent the first one is the coordinate in mapped space, and the second one is the full dimension of the data to get a normalised value.